coalib/parsing/Globbing.py

```diff
@@ -29,6 +29,39 @@ def _end_of_set_index(string, start_index):
     return closing_index
 
 
+def escape_brackets_and_parentheses(string):
+    """
+    Escapes the given string with [c] pattern
+    """
+    # We need to handle '[' and ']' specially
+    tmp = ""
+    prev = 0
+    while True:
+        lb = string[prev:].find("[")
+        rb = string[prev:].find("]")
+        if lb is -1 or rb is -1:
+            cur = max(lb, rb)
+        else:
+            cur = min(lb, rb)
+        if cur is -1:
+            tmp += string[prev:]
+            break
+        cur = cur + prev
+        tmp += string[prev:cur]
+        tmp += "[" + string[cur] + "]"
+        prev = cur + 1
+    return tmp.replace("(", "[(]").replace(")", "[)]")
+
+
+def unescape_brackets_and_parentheses(string):
+    """
+    Unescapes the given string
+    Note: Only for strings escaped with escape_brackets_and_parentheses
+    """
+    string = string.replace("[(]", "(").replace("[)]", ")")
+    return string.replace("[[]", "[").replace("[]]", "]")
```

Comment:
@sils1297 Say there is a path `/home/user/Downloads (user | root)/coala (1)/...`. Then `unescape_brackets_and_parentheses` would escape it as `/home/user/Downloads [(]user | root[)]/coala [(]1[)]/...` which is correct, since this would not be interpreted as alternatives (that part has been taken care of in `_position_is_bracketed` - this way only those `|` inside valid brackets are considered)